x-pack/plugins/security_solution/scripts/endpoint/resolver_generator_script.ts

```diff
@@ -179,35 +220,70 @@ async function main() {
       type: 'boolean',
       default: false,
     },
+    newUser: {
+      alias: 'nu',
+      describe: 'Adds a new user with username:password',
+      type: 'string',
+      default: 'endpoint_user:changeme',
+    },
   }).argv;
   let ca: Buffer;
-  let kbnClient: KbnClient;
+
   let clientOptions: ClientOptions;
+  let url: string;
+  let node: string;
+  const toolingLogOptions = {
+    log: new ToolingLog({
+      level: 'info',
+      writeTo: process.stdout,
+    }),
+  };
+
+  let kbnClientOptions: KbnClientOptions = {
+    ...toolingLogOptions,
+    url: argv.kibana,
+  };
 
   if (argv.ssl) {
     ca = fs.readFileSync(CA_CERT_PATH);
-    const url = argv.kibana.replace('http:', 'https:');
-    const node = argv.node.replace('http:', 'https:');
-    kbnClient = new KbnClient({
-      log: new ToolingLog({
-        level: 'info',
-        writeTo: process.stdout,
-      }),
+    url = argv.kibana.replace('http:', 'https:');
+    node = argv.node.replace('http:', 'https:');
+    kbnClientOptions = {
+      ...kbnClientOptions,
       url,
       certificateAuthorities: [ca],
-    });
+    };
+
     clientOptions = { node, tls: { ca: [ca] } };
   } else {
-    kbnClient = new KbnClient({
-      log: new ToolingLog({
-        level: 'info',
-        writeTo: process.stdout,
-      }),
-      url: argv.kibana,
-    });
     clientOptions = { node: argv.node };
   }
-  const client = new Client(clientOptions);
+  let client = new Client(clientOptions);
+  // add endpoint user
+  const newUserCreds = argv.newUser ? argv.newUser.split(':') : undefined;
+  const user: UserInfo | undefined = await addUser(
+    client,
+    newUserCreds
+      ? {
+          username: newUserCreds[0],
+          password: newUserCreds[1],
+        }
+      : undefined
+  );
+
+  // update client and kibana options before instantiating
+  if (user) {
+    // use endpoint user for Es and Kibana URLs
+    url = argv.kibana.replace('elastic', user.username);
```

Comment:
this (and below) is an assumption that `elastic` was used in this param, so this is no good here. It also does not set the password. We need to understand and document that the `newUser` will do to the other param values if we are going to replace their user:password.

My suggestion (building on my prior comments) would be to a) `newUser` shoudl not have a default. Then it would be up to the person running the CLI tool to ensure that 1) they set that new `newUser` param with their value and adjust the other params to also use it. I think that would avoid you overriding the user's param values in the code.

Also remember that I am not sure that many people run wiht the `--fleet` options aside from our team, so I also think that this new user shoudl only be used if the `--fleet` options is being used

Reply:
I see what you mean. I wasn't aware that there are other teams that use this without the fleet flag. 

So It makes sense to then only add a new user if both `--fleet` is set to true and `--newUser` has values via CLI. I'll go ahead and make that change.

Reply:
I see what you mean. I wasn't aware that there are other teams that use this without the fleet flag.

So It makes sense to then only add a new user if both --fleet is set to true and --newUser has values via CLI. I'll go ahead and make that change.

Reply:
done 5ce10629642bcdf4d08368f8b1f53374601d1046.